crowbar_framework/app/models/cinder_service.rb

```diff
@@ -205,6 +205,20 @@ def apply_role_pre_chef_call(old_role, role, all_nodes)
     # the VIP of the cluster to be setup
     allocate_virtual_ips_for_any_cluster_in_networks(controller_elements, vip_networks)
 
+    # Generate secrets uuid for libvirt rbd backend
+    proposal = ProposalObject.find_data_bag_item "crowbar/bc-cinder-default"
```

Comment:
> Please don't say that it is bad idea unless you have some other proposal how to solve it.

I didn't say it was a bad idea I just said I am unsure and asked a specific question to better understand the problem. One argument against using Javascript here is e.g. that it would making creating proposals through the commandline more complicated as the user would have to generate the uuids manually.

One way to keep doing it inside the rails app while avoiding the issue that @vuntz raised in https://github.com/crowbar/barclamp-cinder/pull/249/files#r28976673 might be to override the ```save_proposal!``` method and generate the UUIDs there.